setup.py

```diff
@@ -34,6 +41,11 @@ def read_injector_variable(name):
 version = read_injector_variable('__version__')
 version_tag = read_injector_variable('__version_tag__')
 
+
+requirements = tuple(obtain_requirements('requirements.txt'))
+requirements_dev = tuple(obtain_requirements('requirements-dev.txt'))
```

Comment:
We don't need to use mutable collection here, that's why I used tuple, it's logical.
I don't know why you are so picky about it.
I'd understand if you said to use a tuple instead of lists but vice versa it doesn't make sense.

Reply:
> A homogeneous collection has nothing to do with it.

It does when you look at how tuples and lists are being used in the wild. Many people (me including) and projects (like this one) use tuples for heterogeneous collections and lists for 
See how `typing.List` can only have one type parameter covering all elements of the list but `typing.Tuple` is variadic and has one type parameter per item (https://docs.python.org/3/library/typing.html#typing.Tuple)

Reply:
>It does when you look at how tuples and lists are being used in the wild.

Thanks for the detailed reply.
In Python, List and Tuple are more flexible even according to the official docs.
https://docs.python.org/3/library/stdtypes.html#tuples ("Tuples are also used for cases where an immutable sequence of homogeneous data is needed")

>See how typing.List can only have one type parameter 

Yes and in the official doc there is an example of how to get over it using TypeVar (https://docs.python.org/3/library/typing.html#typing.List)

My point is Python is more flexible to think about `homogenous/heterogenous` so strictly like you.

I can follow your convention, no problem but what is the benefit of it?
For example, if I use lists only when I need to have a mutable collection then I save resources and protect data more often. That is the real benefit of such an approach.

Reply:
I'm not asking you to change your conviction or anything. The only distinction that matters here is vaguely-defined elegance and all things being equal (we all have just opinions here) I'm willing to stick with mine in the context of this project.

If we had a case where a tuple would provide significant resource or safety benefits over a list (demonstrated with some evidence) it'd be a different story.

PS. The `TypeVar` example won't allow you to mix `int` and `float` within the same list, it's still a homogeneous collection.

Reply:
>PS. The TypeVar example won't allow you to mix int and float within the same list, it's still a homogeneous collection.

```
from typing import TypeVar, List


T = TypeVar('T', int, float)


def vec2(x: T, y: T) -> List[T]:
    return [x, y]


assert vec2(3, 3.0) == [3, 3.0]
```

Reply:
I *obviously* meant this in a static type context (as verified by mypy or something similar), not in runtime context (where `TypeVar` has no significance). And admittedly `int` and `float` is not a good example, since mypy considers `float` to be a supertype of `int`[1]. But test this with mypy or anything else:

```python
# qwe.py
from typing import TypeVar, List


T = TypeVar('T', int, str)


def vec2(x: T, y: T) -> List[T]:
    return [x, y]

vec2(1, 'asd')
```

```
% mypy qwe.py
qwe.py:10: error: Value of type variable "T" of "vec2" cannot be "object"
```

[1] https://mail.python.org/archives/list/typing-sig@python.org/message/JWUTJGOL6FOXY5RJOZJWQEUXWTO5NIBO/

Reply:
>I obviously meant this in a static type context 

I checked it via `mypy` and it didn't find any errors, that's why I posted it.

Reply:
Apologies, I missed that from your response (but yeah, `int` and `float` is the unfortunate corner-case that doesn't illustrate the case well with mypy).